Bump tasks assembly and file versions automatically with every nightly build

Currently both are hard-coded to 1.0.0.0.

We keep running in to issues where there is a build-to-build break in the SDK and an msbuild process can't SxS load two SDK task assemblies with the same name.

I think we can bump both with every daily build number. It's a simple fix and since the tasks assembly is not designed to have any exchange types, SxS loading any two different builds should be acceptable.

cc @davkean @rainersigwald @dsplaisted

@srivatsn That [commit](https://github.com/dotnet/sdk/pull/616/commits/33504deed0c147f9a21e2ce3c7fd38c5276cf7f7#diff-c1fa1a45a54e9930158c395d92afe031R34) caused Microbuild failure with the below MSBuild error:

```
2017-01-10T01:15:24.2998766Z E:\A\_work\15\s\Common.props(32,5): error MSB4186: Invalid static method invocation syntax: "[MSBuild]::Subtract($(BuildNumber.Split('.')[0].Substring(3).Trim()), 8800)". Input string was not in a correct format. Static method invocation should be of the form: $([FullTypeName]::Method()), e.g. $([System.IO.Path]::Combine(`a`, `b`)).  [E:\A\_work\15\s\build\build.proj]
2017-01-10T01:15:24.4717498Z Failed to build
```

This passed fine locally if I set BUILD_BUILDNUMBER, which forces this invocation and also generates correct build numbers. I tried both on a machine with just Dev14 as well as one with just Dev15 and locally it works fine. We use the exact same syntax in Roslyn too and it works fine: https://github.com/dotnet/roslyn/blob/master/build/Targets/Versions.props#L43. @srivatsn you know if there is anything special about the microbuild machine for SDK repo?

@rainersigwald you have any suggestions how to address this failure?

Setting GenerateAssemblyInfo to true for Microsoft.NET.Build.Tasks.csproj gives the following errors:

C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(6,12): error CS0579: Duplicate 'System.Reflection.AssemblyConfigurationAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(7,12): error CS0579: Duplicate 'System.Reflection.AssemblyCopyrightAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(8,12): error CS0579: Duplicate 'System.Reflection.AssemblyDescriptionAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(9,12): error CS0579: Duplicate 'System.Reflection.AssemblyFileVersionAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(11,12): error CS0579: Duplicate 'System.Reflection.AssemblyProductAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(12,12): error CS0579: Duplicate 'System.Reflection.AssemblyTitleAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(13,12): error CS0579: Duplicate 'System.Reflection.AssemblyVersionAttribute' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]
C:\sdk\bin\obj\Microsoft.NET.Build.Tasks\Debug\netstandard1.3\Microsoft.NET.Build.Tasks.AssemblyInfo.cs(14,12): error CS0579: Duplicate 'System.Resources.NeutralResourcesLanguage' attribute [C:\sdk\src\Tasks\Microsoft.NET.Build.Tasks\Microsoft.NET.Build.Tasks.csproj]


Additionally, it seems none of the template assemblies generated in the output dir, i.e. Microsoft.VisualStudio.ProjectSystem.XXX.Templates.dll, respect this flag and always have all versions 0.0.0.0.

I have pushed all the changes I made locally to https://github.com/mavasani/sdk/blob/BuildVersion/Common.props. @nguerrera I am moving over the bug to you.

--- a/Common.props
+++ b/Common.props
@@ -50,7 +50,8 @@
     <!-- Prepare assembly metadata -->
     <Authors>Microsoft Corporation</Authors>
     <Copyright>© Microsoft Corporation. All rights reserved.</Copyright>
-    <AssemblyVersion>$(VersionPrefix).0</AssemblyVersion>
+    <AssemblyVersion Condition="'$(BUILD_BUILDNUMBER)' == ''">$(VersionPrefix).0</AssemblyVersion>
+    <AssemblyVersion Condition="'$(BUILD_BUILDNUMBER)' != ''">$(VersionPrefix).$([MSBuild]::Subtract($(BuildNumberPart1), $(BuildEpoch)))</AssemblyVersion>
     <FileVersion Condition="'$(BUILD_BUILDNUMBER)' == ''">$(VersionPrefix).0</FileVersion>
     <FileVersion Condition="'$(BUILD_BUILDNUMBER)' != ''">$(VersionPrefix).$([MSBuild]::Subtract($(BuildNumberPart1), $(BuildEpoch)))</FileVersion>
     <InformationalVersion>$(VersionPrefix).$(BuildNumber)</InformationalVersion>

